### PR TITLE
WIP: Reactstrap Header

### PR DESCRIPTION
**Pull request by vikr01**
#### Pre-Submission Checklist

- [ ] Your pull request targets the `master` branch.
- [x] Branch starts with either `fix/` or `feature/` (e.g. `fix/signin-issue`)
- [x] All new and existing tests pass.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description

- Header has been rewritten with ReactStrap components.
- Utilized `SyntheticEvent` type from React to fix flow errors on cases of `Event<T>`.

### CODE REVIEW

**Comment by amilajack** [comment]
Nice! Can you share screenshots of all the pages that were changed?

File: app/components/header/Header.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+  Collapse,
+  Form,
+  Input,
+  Nav,
+  Navbar,
+  NavbarToggler,
+  NavItem
+} from 'reactstrap';
 import React, { Component } from 'react';
+import type { Node, SyntheticEvent as Event } from 'react';
```
Comment (amilajack) [inline app/components/header/Header.js]: type imports should go before relative imports. relative type imports should go after relative imports

File: app/components/header/Header.js [hunk 2]
```diff
@@ -31,24 +42,24 @@ export default class Header extends Component {
   /**
    * Set the mode of the movies to be 'search'
    */
-  setSearchState(searchQuery: string) {
+  setSearchState(searchQuery: string): void {
```
Comment (amilajack) [inline app/components/header/Header.js]: `: void` annotation is unnecessary

**Comment by vikr01** [comment]
Fixed the type imports and removed the void declarations. No pages were changed. Though, this isn't really utilizing React features -- it's just using the corresponding component from `reactstrap` instead of the html element.

**Pull request merged**